### PR TITLE
refactor credential endpoints

### LINKED ISSUE #3644: Refactor credential endpoints
**User Story**
Create one endpoint for credential names for providers

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: Create one endpoint for credential names for providers

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3644

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/credentials/manager.go [hunk 1]
```diff
@@ -118,6 +118,10 @@ func New() *Manager {
 	return &Manager{credentials: credentials}
 }
 
+func NewWithCredentials(credentials *Credentials) *Manager {
+	return &Manager{credentials: credentials}
+}
+
 // NewFromFiles returns a instance of manager with the credentials loaded from the given paths
 func NewFromFiles(credentialsFilename string) (*Manager, error) {
```
Comment (floreks) [inline api/pkg/credentials/manager.go]: nit: Shouldn't this be `NewFromFile`?

File: api/pkg/handler/v1/credentials/credentials.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		err := req.Validate()
+		if err != nil {
+			return nil, errors.NewBadRequest(err.Error())
+		}
+
+		credentials := apiv1.CredentialList{}
+		names := make([]string, 0)
+
+		switch req.ProviderName {
+		case "aws":
```
Comment (floreks) [inline api/pkg/handler/v1/credentials/credentials.go]: Could you move those to enum?

Comment (zreigz) [reply api/pkg/handler/v1/credentials/credentials.go]: yes

**Comment by floreks** [comment]
Nice!

/lgtm
/approve

**Pull request merged**